mathlibtools/leanproject.py

```diff
@@ -204,7 +179,17 @@ def get_project(name: str, new_branch: bool, directory: str = '') -> None:
         LeanProject.from_git_url(url, directory, branch, new_branch,
                                  cache_url, force_download)
     except GitCommandError as err:
-        handle_exception(err, 'Git command failed')
+        # if full url is provided, do not retry with HTTPS
+        if ('Permission denied' in err.stderr) and not is_url:
```

Comment:
This is typical USA/UK programming... What about users whose locale is not English?